torch/_functorch/partitioners.py

```diff
@@ -804,16 +804,20 @@ def solve_min_cut(
             if node.op == "call_function" and hasattr(node.target, "_overloadpacket")
         }
         ops_ignored = joint_module_ops - {str(i) for i in op_types.recomputable_ops}
-        print("Ops banned from rematerialization: ", ops_ignored)
+        print("Ops banned from re-materialization: ", ops_ignored)
         print()
 
     def can_fuse_into_auto_functionalized(a, b):
-        if b.target != torch.ops.higher_order.auto_functionalized:
+        if (
+            b.target != torch.ops.higher_order.auto_functionalized
+            and b.target != torch.ops.higher_order.auto_functionalized_v2
+        ):
             return False
         mutable_op = b.args[0]
-        mutable_arg_names = (
-            torch._higher_order_ops.auto_functionalize.get_mutable_arg_names(mutable_op)
-        )
+        (
+            mutable_arg_names,
+            _,
+        ) = torch._higher_order_ops.auto_functionalize.get_mutable_args(mutable_op)
```

Comment:
I'm not completely sure about this interaction.

Reply:
I think this should be "we can fuse a node X into auto_functionalized if X is one of the bases". The code as written is "we can fuse a node X into auto_functionalized if X is one of the mutated args". Not completely sure though, doesn't need to happen in this PR

Reply:
do we fuse before decompose?

Reply:
I probably need to file issue for all those follow ups. or create an issue with a list